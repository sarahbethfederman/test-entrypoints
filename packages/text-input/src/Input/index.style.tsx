import styled, { css } from 'styled-components';
import { deriveSize, normalise } from '@lendi-ui/utils';
import { select } from '@lendi-ui/theme';
import { map, BreakpointValue, BreakpointValueMap } from '@lendi-ui/breakpoint';
import { InputButton } from '../InputButton/index';

type Size = 'sm' | 'md' | 'lg';
export type InputSize = BreakpointValue<Size> | BreakpointValueMap<Size>;

const heightBySizeMixin = (size: InputSize) =>
  map(size, (val) => {
    switch (val) {
      case 'sm':
        return `
          height: ${deriveSize(2.5)};
      `;
      case 'md':
        return `
          height: ${deriveSize(3)};
      `;
      case 'lg':
        return `
          height: ${deriveSize(4)};
      `;
      default:
        return undefined;
    }
  });

const fontBySizeMixin = (size: InputSize) =>
  map(size, (val) => {
    switch (val) {
      case 'sm':
        return `
        font-size: ${deriveSize(1)};
    `;
      case 'md':
        return `
        font-size: ${deriveSize(1.125)};
    `;
      case 'lg':
        return `
        font-size: ${deriveSize(1.375)};
    `;
      default:
        return undefined;
    }
  });

const InputBorderColor = ({ isInverse, isError }: LayoutProp) => {
  let color;
  if (!isError) {
    if (!isInverse) {
      color = select('colors.shade.200');
    } else {
      color = select('colors.shade.25');
    }
  } else {
    color = select('colors.error.500');
  }
  return color;
};

const InputBorderHoverColor = ({ isInverse, isError }: LayoutProp) => {
  let color;
  if (!isError) {
    if (!isInverse) {
      color = select('colors.shade.400');
    } else {
      color = select('colors.shade.0');
    }
  } else {
    color = select('colors.error.500');
  }
  return color;
};

export interface LayoutProp {
  isFullWidth: boolean;
  size: InputSize;
  isInverse: boolean;
  isError: boolean;
  isDisabled: boolean;
}
export const Layout = styled.div`
  ${normalise};
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  border-radius: ${select('borderRadius')};
  box-sizing: border-box;
  ${({ isFullWidth, size, isInverse, isDisabled }: LayoutProp) => {
    if (isDisabled) {
      return css`
        cursor: not-allowed;
        width: ${isFullWidth ? '100%' : 'auto'};
        background-color: ${isInverse ? 'transparent' : select('colors.shade.25')};
        border: 1px solid ${isInverse ? select('colors.shade.0') : select('colors.shade.100')};
        ${heightBySizeMixin(size)};
      `;
    } else {
      return css`
        border: 1px solid ${InputBorderColor};
        width: ${isFullWidth ? '100%' : 'auto'};
        background-color: ${isInverse ? 'transparent' : select('colors.shade.0')};
        ${heightBySizeMixin(size)};
      `;
    }
  }};

  :hover:not(:focus) {
    border: 1px solid ${InputBorderHoverColor};
  }
`;

export interface InputWrapperProps {
  fontSize: InputSize;
  isInverse: boolean;
  isError: boolean;
  isDisabled?: boolean;
}

export const InputWrapper = styled.input`
  height: 100%;
  min-width: 0;
  width: 100%;
  border: 0px;
  border-radius: ${select('borderRadius')};
  ${({ fontSize, isInverse }: InputWrapperProps) => css`
    ${fontBySizeMixin(fontSize)}
    color: ${isInverse ? select('colors.shade.0') : select('colors.shade.700')};
    background-color: ${isInverse ? 'transparent' : select('colors.shade.0')};
    font-family: ${select('typography.body.fontFamily')};
    padding: 0 ${deriveSize(0.75)};
  `} ::placeholder {
    color: ${select('colors.shade.300')};
  }

  :disabled {
    cursor: not-allowed;
    ${({ isInverse }: InputWrapperProps) => css`
      background-color: ${isInverse ? 'transparent' : select('colors.shade.25')};
    `};
  }
`;

export const BeforeWrapper = styled.span`
  ${InputButton} {
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
    margin: 0 0 0 -1px;
  }
  padding-right: 1px;
  ${({ isDisabled }: { isDisabled: boolean }) => {
    if (isDisabled) {
      return css`
        pointer-events: none;
      `;
    }
    return undefined;
  }};
`;

export const AfterWrapper = styled.span`
  ${InputButton} {
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
    margin: 0 -1px 0 0;
  }
  padding-left: 1px;
  ${({ isDisabled }: { isDisabled: boolean }) => {
    if (isDisabled) {
      return css`
        pointer-events: none;
      `;
    }
    return undefined;
  }};
`;
