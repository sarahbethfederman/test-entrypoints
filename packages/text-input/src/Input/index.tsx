import * as React from 'react';
import { InputSize, InputWrapper, Layout, BeforeWrapper, AfterWrapper } from './index.style';

export interface InputProp {
  size?: InputSize;
  isError?: boolean;
  isInverse?: boolean;
  isDisabled?: boolean;
  isFullWidth?: boolean;
  before?: React.ReactNode;
  after?: React.ReactNode;
  placeholder?: string;
  value?: string;
  onFocus?: (event: React.FocusEvent<HTMLInputElement>) => void;
  onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

export const Input = ({
  size = 'md',
  before,
  after,
  isError = false,
  isInverse = false,
  isFullWidth = false,
  isDisabled = false,
  ...otherProps
}: InputProp) => {
  return (
    <Layout size={size} isFullWidth={isFullWidth} isInverse={isInverse} isError={isError} isDisabled={isDisabled}>
      {before && <BeforeWrapper isDisabled={isDisabled}>{before}</BeforeWrapper>}
      <InputWrapper
        type="text"
        readOnly={isDisabled}
        isError={isError}
        isInverse={isInverse}
        fontSize={size}
        {...otherProps}
      />
      {after && <AfterWrapper isDisabled={isDisabled}>{after}</AfterWrapper>}
    </Layout>
  );
};
