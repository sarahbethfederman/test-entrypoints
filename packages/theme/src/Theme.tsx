import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
