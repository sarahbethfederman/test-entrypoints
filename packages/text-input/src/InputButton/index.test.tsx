import * as React from 'react';
import { mount } from 'enzyme';
import Theme from '@lendi-ui/theme';
import { InputButton } from './index';
import { Input } from '../Input/index';

let element;

function render(props) {
  element = mount(
    <Theme>
      <Input size="md" placeholder="input here ..." onChange={() => {}} isFullWidth={true} {...props} />
    </Theme>
  );
}

describe('InputButton', () => {
  it('it should render InputButton in before wrapper', () => {
    const before = (
      <InputButton variant="primary" size="md">
        search
      </InputButton>
    );
    render({ before });
    expect(element.find(InputButton)).toMatchSnapshot();
  });

  it('it should render InputButton in after wrapper', () => {
    const after = (
      <InputButton variant="primary" size="md">
        search
      </InputButton>
    );
    render({ after });
    expect(element.find(InputButton)).toMatchSnapshot();
  });
});
