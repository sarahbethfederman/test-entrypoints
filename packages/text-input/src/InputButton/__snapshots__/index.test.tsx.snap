// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputButton it should render InputButton in after wrapper 1`] = `
.c2 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  cursor: pointer;
  box-sizing: border-box;
  border: none;
  border-radius: 6px;
  font-weight: 600;
  line-height: 1;
  text-align: center;
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
  text-transform: uppercase;
  height: calc(3 * var(--lendi-ui-size));
  padding: 0 var(--lendi-ui-size);
  font-size: calc(0.875 * var(--lendi-ui-size));
  -webkit-letter-spacing: calc(0.0625 * var(--lendi-ui-size));
  -moz-letter-spacing: calc(0.0625 * var(--lendi-ui-size));
  -ms-letter-spacing: calc(0.0625 * var(--lendi-ui-size));
  letter-spacing: calc(0.0625 * var(--lendi-ui-size));
  font-family: 'Open Sans',sans-serif;
  color: #ffffff;
  background-color: #00c0a5;
}

.c2:focus {
  outline: none;
  box-shadow: 0 3px 3px 0 rgba(0,0,0,0.1), 0 3px 4px 0 rgba(0,0,0,0.12), 0 1px 8px 0 rgba(0,0,0,0.14);
}

.c2:hover {
  box-shadow: 0 12px 17px 2px rgba(0,0,0,0.1), 0 5px 22px 4px rgba(0,0,0,0.12), 0 7px 8px 0 rgba(0,0,0,0.14);
}

.c2:active {
  -webkit-transform: scale(0.98);
  -ms-transform: scale(0.98);
  transform: scale(0.98);
}

.c2:focus {
  background-color: #32ccb7;
}

.c2:hover {
  background-color: #32ccb7;
}

.c2:active {
  background-color: #009984;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  margin-right: var(--lendi-ui-size);
}

.c5 {
  margin-left: var(--lendi-ui-size);
}

.c1:focus {
  outline: 5px auto -webkit-focus-ring-color;
  opacity: 1;
  box-shadow: none;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c1:active {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c6 .c0 {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  margin: 0 0 0 -1px;
}

.c7 .c0 {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  margin: 0 -1px 0 0;
}

<Styled(InputWithButton)
  size="md"
  variant="primary"
>
  <InputWithButton
    className="c0 c1"
    size="md"
    variant="primary"
  >
    <Styled(Button)
      className="c0 c1"
      size="md"
      variant="primary"
    >
      <Button
        className="c0 c1 "
        size="md"
        variant="primary"
      >
        <styled.button
          aria-label="search"
          className="c0 c1 "
          size="md"
          variant="primary"
        >
          <button
            aria-label="search"
            className="c0 c1 c2"
            size="md"
          >
            <styled.div>
              <div
                className="c3"
              >
                <styled.span
                  size="md"
                >
                  <span
                    className="c4"
                    size="md"
                  />
                </styled.span>
                <span>
                  search
                </span>
                <styled.span
                  size="md"
                >
                  <span
                    className="c5"
                    size="md"
                  />
                </styled.span>
              </div>
            </styled.div>
          </button>
        </styled.button>
      </Button>
    </Styled(Button)>
  </InputWithButton>
</Styled(InputWithButton)>
`;

exports[`InputButton it should render InputButton in before wrapper 1`] = `
.c2 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  cursor: pointer;
  box-sizing: border-box;
  border: none;
  border-radius: 6px;
  font-weight: 600;
  line-height: 1;
  text-align: center;
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
  text-transform: uppercase;
  height: calc(3 * var(--lendi-ui-size));
  padding: 0 var(--lendi-ui-size);
  font-size: calc(0.875 * var(--lendi-ui-size));
  -webkit-letter-spacing: calc(0.0625 * var(--lendi-ui-size));
  -moz-letter-spacing: calc(0.0625 * var(--lendi-ui-size));
  -ms-letter-spacing: calc(0.0625 * var(--lendi-ui-size));
  letter-spacing: calc(0.0625 * var(--lendi-ui-size));
  font-family: 'Open Sans',sans-serif;
  color: #ffffff;
  background-color: #00c0a5;
}

.c2:focus {
  outline: none;
  box-shadow: 0 3px 3px 0 rgba(0,0,0,0.1), 0 3px 4px 0 rgba(0,0,0,0.12), 0 1px 8px 0 rgba(0,0,0,0.14);
}

.c2:hover {
  box-shadow: 0 12px 17px 2px rgba(0,0,0,0.1), 0 5px 22px 4px rgba(0,0,0,0.12), 0 7px 8px 0 rgba(0,0,0,0.14);
}

.c2:active {
  -webkit-transform: scale(0.98);
  -ms-transform: scale(0.98);
  transform: scale(0.98);
}

.c2:focus {
  background-color: #32ccb7;
}

.c2:hover {
  background-color: #32ccb7;
}

.c2:active {
  background-color: #009984;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  margin-right: var(--lendi-ui-size);
}

.c5 {
  margin-left: var(--lendi-ui-size);
}

.c1:focus {
  outline: 5px auto -webkit-focus-ring-color;
  opacity: 1;
  box-shadow: none;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c1:active {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c6 .c0 {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  margin: 0 0 0 -1px;
}

<Styled(InputWithButton)
  size="md"
  variant="primary"
>
  <InputWithButton
    className="c0 c1"
    size="md"
    variant="primary"
  >
    <Styled(Button)
      className="c0 c1"
      size="md"
      variant="primary"
    >
      <Button
        className="c0 c1 "
        size="md"
        variant="primary"
      >
        <styled.button
          aria-label="search"
          className="c0 c1 "
          size="md"
          variant="primary"
        >
          <button
            aria-label="search"
            className="c0 c1 c2"
            size="md"
          >
            <styled.div>
              <div
                className="c3"
              >
                <styled.span
                  size="md"
                >
                  <span
                    className="c4"
                    size="md"
                  />
                </styled.span>
                <span>
                  search
                </span>
                <styled.span
                  size="md"
                >
                  <span
                    className="c5"
                    size="md"
                  />
                </styled.span>
              </div>
            </styled.div>
          </button>
        </styled.button>
      </Button>
    </Styled(Button)>
  </InputWithButton>
</Styled(InputWithButton)>
`;
